components/Dialogs/src/MDCDialogPresentationController.m

```diff
@@ -146,6 +138,20 @@ - (void)presentationTransitionWillBegin {
   // presentation and dismissal of the dialog.
   // https://material.io/guidelines/motion/choreography.html#choreography-creation
 
+  // Ensure corner radius matches between the tracking view and the view being presented
+  if (self.dialogCornerRadius <= kPresentedViewCorenerRadius) {
+    // If dialogCornerRadius is not set, use the presented view's cornerRadius for the shadow layer.
+    _trackingView.layer.cornerRadius = self.presentedView.layer.cornerRadius;
+  } else {
+    // If dialogCornerRadius is set, use its value for the shadow layer as well as the presented
+    // view layer, overriding any direct assignments to the presented view's cornerRadius.
+    _trackingView.layer.cornerRadius = self.dialogCornerRadius;
+    // Note: For MDCAlertController, this assumes that the "view" property points to the same
+    // instance as the "alertView" property. Therefore, we are safe to not set its cornerRadius
+    // property (its ".cornerRadius", rather then its "presentedView.layer.cornerRadius")
+    self.presentedView.layer.cornerRadius = self.dialogCornerRadius;
```

Comment:
If we want to be extremely cautious here, we could store the previousPresentedViewCornerRadius before overwriting it.  When we stop presenting the view, we could set self.presentedView.layer.cornerRadius back to its original value.

Reply:
Sounds good.